pyccl/halos/profiles_cib.py

```diff
@@ -0,0 +1,316 @@
+from .profiles import HaloProfile, HaloProfileNFW
+from .profiles_2pt import Profile2pt
+from .concentration import Concentration
+import numpy as np
+from scipy.integrate import simps
+from scipy.special import lambertw
+
+
+class HaloProfileCIBShang12(HaloProfile):
+    """ CIB profile implementing the model by Shang et al.
+    (2012MNRAS.421.2832S).
+
+    The parametrization for the mean profile is:
+
+    .. math::
+        j_\\nu(r) = \\frac{1}{4\\pi}
+        \\left(L^{\\rm cen}_{\\nu(1+z)}(M)+
+        L^{\\rm sat}_{\\nu(1+z)}u_{\\rm sat}(r|M)
+        \\right),
+
+    where the luminosity from centrals and satellites is
+    modelled as:
+
+    .. math::
+        L^{\\rm cen}_{\\nu}(M) = L^{\\rm gal}_\\nu(M)\\,
+        N_{\\rm cen}(M),
+
+    .. math::
+        L^{\\rm sat}_{\\nu}(M) = \\int_{M_{\\rm min}}^{M} dm
+        \\frac{dN_{\\rm sub}}{dm}\\,L^{\\rm gal}_\\nu(m).
+
+    Here, :math:`dN_{\\rm sub}/dm` is the subhalo mass function,
+    :math:`u_{\\rm sat}` is the satellite galaxy density profile
+    (modelled as a truncated NFW profile), and the infrared
+    galaxy luminosity is parametrized as
+
+    .. math::
+        L^{\\rm gal}_{\\nu}(M,z)=L_0(1+z)^{s_z}\\,
+        \\Sigma(M)\\,S_\\nu,
+
+    where the mass dependence is lognormal
+
+    .. math::
+        \\Sigma(M) = \\frac{M}{\\sqrt{2\\pi\\sigma_{LM}^2}}
+        \\exp\\left[-\\frac{\\log_{10}^2(M/M_{\\rm eff})}
+        {2\\sigma_{LM}^2}\\right],
+
+    and the spectrum is a modified black-body law
+
+    .. math::
+        S_\\nu \\propto\\left\\{
+        \\begin{array}{cc}
+           \\nu^\\beta\\,B_\\nu(T_d) & \\nu < \\nu_0 \\\\
+           \\nu^\\gamma & \\nu \\geq \\nu_0
+        \\end{array}
+        \\right.,
+
+    with the normalization fixed by :math:`S_{\\nu_0}=1`,
+    and :math:`\\nu_0` defined so the spectrum has a continuous
+    derivative for all :math:`\\nu`.
+
+    Finally, the dust temperature is assumed to have a redshift
+    dependence of the form :math:`T_d=T_0(1+z)^\\alpha`.
+
+    Args:
+        c_M_relation (:obj:`Concentration`): concentration-mass
+            relation to use with this profile.
+        nu_GHz (float): frequency in GHz.
+        alpha (float): dust temperature evolution parameter.
+        T0 (float): dust temperature at :math:`z=0` in Kelvin.
+        beta (float): dust spectral index.
+        gamma (float): high frequency slope.
+        s_z (float): luminosity evolution slope.
+        log10meff (float): log10 of the most efficient mass.
+        sigLM (float): logarithmic scatter in mass.
+        Mmin (float): minimum subhalo mass.
+        L0 (float): luminosity scale (in
+            :math:`{\\rm Jy}\\,{\\rm Mpc}^2\\,M_\\odot^{-1}`).
+    """
+    one_over_4pi = 0.07957747154
+
```

Comment:
might be useful to give these profiles a name, like in all other halo profiles

Reply:
also, `one_over_4pi` could become a private attribute as it is only stored here to save lines of code and doesn't serve any other useful function within the profile

Reply:
both good points! Done.